<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /local/users/florent/devel/hpp/install/share/agimus_demos/franka/manipulation/urdf/demo.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="pandas">
  <link name="support_link">
    <inertial>
      <mass value="200"/>
      <origin xyz="0 0 0.45"/>
      <inertia ixx="50" ixy="0" ixz="0" iyy="50" iyz="0" izz="50"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0.38"/>
      <geometry>
        <box size="1.703 0.803 0.76"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0.38"/>
      <geometry>
        <box size="1.703 0.803 0.76"/>
      </geometry>
    </collision>
    <!--visual>
          <origin xyz="-0.61 0.2215 0.77"/>
          <geometry>
	    <box size="0.363 0.252 0.02"/>
          </geometry>
        </visual>
        <collision>
          <origin xyz="-0.61 0.2215 0.77"/>
          <geometry>
	    <box size="0.363 0.252 0.02"/>
          </geometry>
        </collision-->
    <visual>
      <origin xyz="0.6385 -0.1655 0.77"/>
      <geometry>
        <box size="0.3 0.3 0.02"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0.6385 -0.1655 0.77"/>
      <geometry>
        <box size="0.3 0.3 0.02"/>
      </geometry>
    </collision>
  </link>
  <link name="wall_link">
    <inertial>
      <mass value="200"/>
      <origin xyz="0 0 1."/>
      <inertia ixx="50" ixy="0" ixz="0" iyy="50" iyz="0" izz="50"/>
    </inertial>
    <visual>
      <origin xyz="0 0 1.0"/>
      <geometry>
        <box size="3.0 0.1 2.0"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 1.0"/>
      <geometry>
        <box size="3.0 0.1 2.0"/>
      </geometry>
    </collision>
  </link>
  <!-- Wrap panda1 robot into a box shaped obstacle -->
  <link name="panda1_box">
    <inertial>
      <mass value="100"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="25" ixy="0" ixz="0" iyy="25" iyz="0" izz="25"/>
    </inertial>
    <collision>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.44 0.803 0.80"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.44 0.803 0.80"/>
      </geometry>
      <material name="lightGrey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
    </visual>
  </link>
  <!-- Gazebo requires a joint to a link called "world" for statically mounted robots -->
  <link name="world"/>
  <joint name="world_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="support_link"/>
  </joint>
  <joint name="support_joint" type="fixed">
    <origin rpy="0 0 3.141592653589793" xyz="0.563 -0.1655 .78"/>
    <parent link="support_link"/>
    <child link="panda2_link0"/>
  </joint>
  <joint name="wall_joint" type="fixed">
    <origin xyz="0 1.07 0"/>
    <parent link="world"/>
    <child link="wall_link"/>
  </joint>
  <joint name="panda1_box_joint" type="fixed">
    <origin xyz="-0.6315 0 1.17"/>
    <parent link="world"/>
    <child link="panda1_box"/>
  </joint>
  <!-- sub-link defining detailed meshes for collision with environment -->
  <link name="panda2_link0">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link0.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link0.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- sub-link defining coarse geometries of real robot's internal self-collision -->
  <link name="panda2_link0_sc">
    <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="-0.075 0 0.06"/>
      <geometry>
        <cylinder length="0.03" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="-0.06 0 0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="-0.09 0 0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <!-- fixed joint between both sub-links -->
  <joint name="panda2_link0_sc_joint" type="fixed">
    <origin rpy="0 0 0"/>
    <parent link="panda2_link0"/>
    <child link="panda2_link0_sc"/>
  </joint>
  <!-- sub-link defining detailed meshes for collision with environment -->
  <link name="panda2_link1">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link1.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link1.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- sub-link defining coarse geometries of real robot's internal self-collision -->
  <link name="panda2_link1_sc">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.1915"/>
      <geometry>
        <cylinder length="0.283" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.05000000000000002"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.33299999999999996"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <!-- fixed joint between both sub-links -->
  <joint name="panda2_link1_sc_joint" type="fixed">
    <origin rpy="0 0 0"/>
    <parent link="panda2_link1"/>
    <child link="panda2_link1_sc"/>
  </joint>
  <joint name="panda2_joint1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.333"/>
    <parent link="panda2_link0"/>
    <child link="panda2_link1"/>
    <axis xyz="0 0 1"/>
    <limit effort="87.0" lower="-2.8973" upper="2.8973" velocity="2.175"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <dynamics D="1" K="7000" damping="0.003" friction="0.0" mu_coulomb="0" mu_viscous="16"/>
  </joint>
  <!-- sub-link defining detailed meshes for collision with environment -->
  <link name="panda2_link2">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link2.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link2.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- sub-link defining coarse geometries of real robot's internal self-collision -->
  <link name="panda2_link2_sc">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.12" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <!-- fixed joint between both sub-links -->
  <joint name="panda2_link2_sc_joint" type="fixed">
    <origin rpy="0 0 0"/>
    <parent link="panda2_link2"/>
    <child link="panda2_link2_sc"/>
  </joint>
  <joint name="panda2_joint2" type="revolute">
    <origin rpy="-1.5707963267948966 0 0" xyz="0 0 0"/>
    <parent link="panda2_link1"/>
    <child link="panda2_link2"/>
    <axis xyz="0 0 1"/>
    <limit effort="87.0" lower="-1.7628" upper="1.7628" velocity="2.175"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-1.7628" soft_upper_limit="1.7628"/>
    <dynamics D="1" K="7000" damping="0.003" friction="0.0" mu_coulomb="0" mu_viscous="16"/>
  </joint>
  <!-- sub-link defining detailed meshes for collision with environment -->
  <link name="panda2_link3">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link3.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link3.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- sub-link defining coarse geometries of real robot's internal self-collision -->
  <link name="panda2_link3_sc">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.145"/>
      <geometry>
        <cylinder length="0.15" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.06999999999999999"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.21999999999999997"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <!-- fixed joint between both sub-links -->
  <joint name="panda2_link3_sc_joint" type="fixed">
    <origin rpy="0 0 0"/>
    <parent link="panda2_link3"/>
    <child link="panda2_link3_sc"/>
  </joint>
  <joint name="panda2_joint3" type="revolute">
    <origin rpy="1.5707963267948966 0 0" xyz="0 -0.316 0"/>
    <parent link="panda2_link2"/>
    <child link="panda2_link3"/>
    <axis xyz="0 0 1"/>
    <limit effort="87.0" lower="-2.8973" upper="2.8973" velocity="2.175"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <dynamics D="1" K="7000" damping="0.003" friction="0.0" mu_coulomb="0" mu_viscous="16"/>
  </joint>
  <!-- sub-link defining detailed meshes for collision with environment -->
  <link name="panda2_link4">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link4.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link4.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- sub-link defining coarse geometries of real robot's internal self-collision -->
  <link name="panda2_link4_sc">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.12" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <!-- fixed joint between both sub-links -->
  <joint name="panda2_link4_sc_joint" type="fixed">
    <origin rpy="0 0 0"/>
    <parent link="panda2_link4"/>
    <child link="panda2_link4_sc"/>
  </joint>
  <joint name="panda2_joint4" type="revolute">
    <origin rpy="1.5707963267948966 0 0" xyz="0.0825 0 0"/>
    <parent link="panda2_link3"/>
    <child link="panda2_link4"/>
    <axis xyz="0 0 1"/>
    <limit effort="87.0" lower="-3.0718" upper="-0.0698" velocity="2.175"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-3.0718" soft_upper_limit="-0.0698"/>
    <dynamics D="1" K="7000" damping="0.003" friction="0.0" mu_coulomb="0" mu_viscous="16"/>
  </joint>
  <!-- sub-link defining detailed meshes for collision with environment -->
  <link name="panda2_link5">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link5.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link5.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- sub-link defining coarse geometries of real robot's internal self-collision -->
  <link name="panda2_link5_sc">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.26"/>
      <geometry>
        <cylinder length="0.1" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.21000000000000002"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.31"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0.08" xyz="0 0.08 -0.13"/>
      <geometry>
        <cylinder length="0.14" radius="0.055"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0.08 -0.06"/>
      <geometry>
        <sphere radius="0.055"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0.08 -0.2"/>
      <geometry>
        <sphere radius="0.055"/>
      </geometry>
    </collision>
  </link>
  <!-- fixed joint between both sub-links -->
  <joint name="panda2_link5_sc_joint" type="fixed">
    <origin rpy="0 0 0"/>
    <parent link="panda2_link5"/>
    <child link="panda2_link5_sc"/>
  </joint>
  <joint name="panda2_joint5" type="revolute">
    <origin rpy="-1.5707963267948966 0 0" xyz="-0.0825 0.384 0"/>
    <parent link="panda2_link4"/>
    <child link="panda2_link5"/>
    <axis xyz="0 0 1"/>
    <limit effort="12.0" lower="-2.8973" upper="2.8973" velocity="2.61"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <dynamics D="1" K="7000" damping="0.003" friction="0.0" mu_coulomb="0" mu_viscous="16"/>
  </joint>
  <!-- sub-link defining detailed meshes for collision with environment -->
  <link name="panda2_link6">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link6.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link6.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- sub-link defining coarse geometries of real robot's internal self-collision -->
  <link name="panda2_link6_sc">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.03"/>
      <geometry>
        <cylinder length="0.08" radius="0.08"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 0.010000000000000002"/>
      <geometry>
        <sphere radius="0.08"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.07"/>
      <geometry>
        <sphere radius="0.08"/>
      </geometry>
    </collision>
  </link>
  <!-- fixed joint between both sub-links -->
  <joint name="panda2_link6_sc_joint" type="fixed">
    <origin rpy="0 0 0"/>
    <parent link="panda2_link6"/>
    <child link="panda2_link6_sc"/>
  </joint>
  <joint name="panda2_joint6" type="revolute">
    <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
    <parent link="panda2_link5"/>
    <child link="panda2_link6"/>
    <axis xyz="0 0 1"/>
    <limit effort="12.0" lower="-0.0175" upper="3.7525" velocity="2.61"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-0.0175" soft_upper_limit="3.7525"/>
    <dynamics D="1" K="7000" damping="0.003" friction="0.0" mu_coulomb="0" mu_viscous="16"/>
  </joint>
  <!-- sub-link defining detailed meshes for collision with environment -->
  <link name="panda2_link7">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link7.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link7.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- sub-link defining coarse geometries of real robot's internal self-collision -->
  <link name="panda2_link7_sc">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.01"/>
      <geometry>
        <cylinder length="0.14" radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 0.08"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.060000000000000005"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0.06 0 0.082"/>
      <geometry>
        <cylinder length="0.01" radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0.065 0 0.082"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0.055 0 0.082"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </collision>
  </link>
  <!-- fixed joint between both sub-links -->
  <joint name="panda2_link7_sc_joint" type="fixed">
    <origin rpy="0 0 0.7853981633974483"/>
    <parent link="panda2_link7"/>
    <child link="panda2_link7_sc"/>
  </joint>
  <joint name="panda2_joint7" type="revolute">
    <origin rpy="1.5707963267948966 0 0" xyz="0.088 0 0"/>
    <parent link="panda2_link6"/>
    <child link="panda2_link7"/>
    <axis xyz="0 0 1"/>
    <limit effort="12.0" lower="-2.8973" upper="2.8973" velocity="2.61"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <dynamics D="1" K="7000" damping="0.003" friction="0.0" mu_coulomb="0" mu_viscous="16"/>
  </joint>
  <link name="panda2_link8"/>
  <joint name="panda2_joint8" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.107"/>
    <parent link="panda2_link7"/>
    <child link="panda2_link8"/>
  </joint>
  <joint name="panda2_hand_joint" type="fixed">
    <parent link="panda2_link8"/>
    <child link="panda2_hand"/>
    <origin rpy="0 0 -0.7853981633974483" xyz="0 0 0"/>
  </joint>
  <!-- sub-link defining detailed meshes for collision with environment -->
  <link name="panda2_hand">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/hand.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/hand.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- sub-link defining coarse geometries of real robot's internal self-collision -->
  <link name="panda2_hand_sc">
  </link>
  <!-- fixed joint between both sub-links -->
  <joint name="panda2_hand_sc_joint" type="fixed">
    <origin rpy="0 0 0"/>
    <parent link="panda2_hand"/>
    <child link="panda2_hand_sc"/>
  </joint>
  <!-- Define the hand_tcp frame -->
  <link name="panda2_hand_tcp"/>
  <joint name="panda2_hand_tcp_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.1034"/>
    <parent link="panda2_hand"/>
    <child link="panda2_hand_tcp"/>
  </joint>
  <link name="panda2_leftfinger">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/finger.dae"/>
      </geometry>
    </visual>
    <!-- screw mount -->
    <collision>
      <origin rpy="0 0 0" xyz="0 18.5e-3 11e-3"/>
      <geometry>
        <box size="22e-3 15e-3 20e-3"/>
      </geometry>
    </collision>
    <!-- cartriage sledge -->
    <collision>
      <origin rpy="0 0 0" xyz="0 6.8e-3 2.2e-3"/>
      <geometry>
        <box size="22e-3 8.8e-3 3.8e-3"/>
      </geometry>
    </collision>
    <!-- diagonal finger -->
    <collision>
      <origin rpy="0.5235987755982988 0 0" xyz="0 15.9e-3 28.35e-3"/>
      <geometry>
        <box size="17.5e-3 7e-3 23.5e-3"/>
      </geometry>
    </collision>
    <!-- rubber tip with which to grasp -->
    <collision>
      <origin rpy="0 0 0" xyz="0 7.58e-3 45.25e-3"/>
      <geometry>
        <box size="17.5e-3 15.2e-3 18.5e-3"/>
      </geometry>
    </collision>
  </link>
  <link name="panda2_rightfinger">
    <visual>
      <origin rpy="0 0 3.141592653589793" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/finger.dae"/>
      </geometry>
    </visual>
    <!-- screw mount -->
    <collision>
      <origin rpy="0 0 0" xyz="0 -18.5e-3 11e-3"/>
      <geometry>
        <box size="22e-3 15e-3 20e-3"/>
      </geometry>
    </collision>
    <!-- cartriage sledge -->
    <collision>
      <origin rpy="0 0 0" xyz="0 -6.8e-3 2.2e-3"/>
      <geometry>
        <box size="22e-3 8.8e-3 3.8e-3"/>
      </geometry>
    </collision>
    <!-- diagonal finger -->
    <collision>
      <origin rpy="-0.5235987755982988 0 0" xyz="0 -15.9e-3 28.35e-3"/>
      <geometry>
        <box size="17.5e-3 7e-3 23.5e-3"/>
      </geometry>
    </collision>
    <!-- rubber tip with which to grasp -->
    <collision>
      <origin rpy="0 0 0" xyz="0 -7.58e-3 45.25e-3"/>
      <geometry>
        <box size="17.5e-3 15.2e-3 18.5e-3"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda2_finger_joint1" type="prismatic">
    <parent link="panda2_hand"/>
    <child link="panda2_leftfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 1 0"/>
    <limit effort="100" lower="0.0" upper="0.04" velocity="0.2"/>
    <dynamics damping="0.3"/>
  </joint>
  <joint name="panda2_finger_joint2" type="prismatic">
    <parent link="panda2_hand"/>
    <child link="panda2_rightfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 -1 0"/>
    <limit effort="100" lower="0.0" upper="0.04" velocity="0.2"/>
    <mimic joint="panda2_finger_joint1"/>
    <dynamics damping="0.3"/>
  </joint>
  <link name="panda2_ref_camera_link"/>
  <joint name="panda2_ref_camera_joint" type="fixed">
    <parent link="panda2_hand"/>
    <child link="panda2_ref_camera_link"/>
    <origin rpy="3.14159265                    -1.57079633                    0.0" xyz="0.02                    0.0                    0.045"/>
  </joint>
  <material name="aluminum">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>
  <material name="plastic">
    <color rgba="0.1 0.1 0.1 1"/>
  </material>
  <!-- camera body, with origin at bottom screw mount -->
  <joint name="camera_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="panda2_ref_camera_link"/>
    <child link="camera_bottom_screw_frame"/>
  </joint>
  <link name="camera_bottom_screw_frame"/>
  <joint name="camera_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.010600000000000002 0.0175 0.0125"/>
    <parent link="camera_bottom_screw_frame"/>
    <child link="camera_link"/>
  </joint>
  <link name="camera_link">
    <visual>
      <!-- the mesh origin is at front plate in between the two infrared camera axes -->
      <origin rpy="1.5707963267948966 0 1.5707963267948966" xyz="0.0043 -0.0175 0"/>
      <geometry>
        <mesh filename="package://realsense2_description/meshes/d435.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.0175 0"/>
      <geometry>
        <box size="0.02505 0.09 0.025"/>
      </geometry>
    </collision>
    <inertial>
      <!-- The following are not reliable values, and should not be used for modeling -->
      <mass value="0.072"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.003881243" ixy="0.0" ixz="0.0" iyy="0.000498940" iyz="0.0" izz="0.003879257"/>
    </inertial>
  </link>
  <!-- camera depth joints and links -->
  <joint name="camera_depth_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="camera_link"/>
    <child link="camera_depth_frame"/>
  </joint>
  <link name="camera_depth_frame"/>
  <joint name="camera_depth_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_depth_frame"/>
    <child link="camera_depth_optical_frame"/>
  </joint>
  <link name="camera_depth_optical_frame"/>
  <!-- camera left IR joints and links -->
  <joint name="camera_infra1_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.0 0"/>
    <parent link="camera_link"/>
    <child link="camera_infra1_frame"/>
  </joint>
  <link name="camera_infra1_frame"/>
  <joint name="camera_infra1_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_infra1_frame"/>
    <child link="camera_infra1_optical_frame"/>
  </joint>
  <link name="camera_infra1_optical_frame"/>
  <!-- camera right IR joints and links -->
  <joint name="camera_infra2_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 -0.05 0"/>
    <parent link="camera_link"/>
    <child link="camera_infra2_frame"/>
  </joint>
  <link name="camera_infra2_frame"/>
  <joint name="camera_infra2_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_infra2_frame"/>
    <child link="camera_infra2_optical_frame"/>
  </joint>
  <link name="camera_infra2_optical_frame"/>
  <!-- camera color joints and links -->
  <joint name="camera_color_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.015 0"/>
    <parent link="camera_link"/>
    <child link="camera_color_frame"/>
  </joint>
  <link name="camera_color_frame"/>
  <joint name="camera_color_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_color_frame"/>
    <child link="camera_color_optical_frame"/>
  </joint>
  <link name="camera_color_optical_frame"/>
  <!-- camera support -->
  <link name="panda2_camera_support_link">
    <inertial>
      <mass value="0.010"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="3.141592653589793 0 -1.5707963267948966" xyz="0.022 -0.102 0.036"/>
      <geometry>
        <mesh filename="package://agimus_demos/franka/manipulation/meshes/camera-support.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="3.141592653589793 0 -1.5707963267948966" xyz="0.022 -0.102 0.036"/>
      <geometry>
        <mesh filename="package://agimus_demos/franka/manipulation/meshes/camera-support.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda2_camera_support_joint" type="fixed">
    <parent link="panda2_hand"/>
    <child link="panda2_camera_support_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <!-- wall behind the table -->
</robot>

